---
title: "PERSON Intervention effects"
author: "A. Umanets"
date: '2023-05-08'
output: 
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---

# Plan

\
**Primary aim:** intervention effects on gut microbiome \
**Secondary aim:** baseline prediction for diet induces changes in metabolic markers \
\
\
**I. Differences in microbiome between T0 and T1** \
\
**1. Change in alpha diversity between T0 and T1** \
[DONE] 1.0. Baseline: T0 between diets with disregard to phenotype.
[DONE] 1.1. Pairwise test per subgroup (phenotype_diet) \
[DONE] 1.2. Pairwise test per diet disregarding phenotype (diet) \
[DONE] 1.3. Pairwise test per phenotype disregarding diet (phenotype) \
[DONE] 1.4. GLM as complete model \
            "alpha ~ sex + age + place + Phenotype\*Diet\*Group + (1|Participant)" \
\
\
**2. Change in bacteria composition between T0 and T1** \
Normalization: CSS, rarefaction. \
Level: ASV, Genus. \
Analysis: \
[DONE] 2.0.1. Baseline: T0 dist ~ sex + age + place + Phenotype*Diet \
[DONE] 2.0.2. Baseline: T1 dist ~ sex + age + place + Phenotype*Diet \
[DONE] 2.1.1 Complete model: \
"dist ~ Participant_ID + sex + age + place + Phenotype\*Diet\*Group" \
\
[DONE] 2.1.2 Complete model: without random effect \
"dist ~ sex + age + place + Phenotype\*Diet\*Group" \
\
[DONE] 2.2.1 Phenotype/diet stratified: \
"dist ~ Participant_ID + sex + age + place + Group" \
\
[DONE] 2.2.2 Phenotype/diet stratified: \
"dist ~ sex + age + place + Group" \
\
\
**3. DA of taxa between T0 and T1** \
[DONE] 3.1. Phenotype/diet stratified: (MaAsLin) term - Group, random effect - Participant \
[DONE] 3.2. Phenotype/diet stratified: Paired Wilcox by Group. Normalization CSS & Rarefaction. \
\
\
**II. Correlation between base microbiome (T0) and change in metabolic parameters.** \
\
**1. Paired correlation**
Concept: Correlated ASVs/Genera at T0 with change in metabolic parameters (delta)
between T0 and T1. \
Normalization: CSS & Rarefaction \
Correlation: Spearman \
\
\
**2. Correlation of overall microbial composition and change in metabolic parameters (delta)** \
Normalization: CSS & rare \
Taxonomic levels: ASV & Genera \
2.1. PERMANOVA: The changes in metabolic parameters (delta) as constitutes terms of the
model: \
"dist ~ delta_metab_1 + delta_metab_2 + delta_metab_3..." \
2.2. envfit: The changes in metabolic parameters (delta) fitted on top of ordination. 
\
\
```{r , message=FALSE, warning=FALSE, prompt=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# Load libraries and set the environment 
#-------------------------------------------------------------------------------
set.seed(549794)

# Load libraries 
libs.list <- c("tidyverse", "metagenomeSeq", "qiime2R", "readxl", "phyloseq", 
               "broom", "ggsignif", "lme4", "vegan", "knitr", "Maaslin2", 
               "ComplexHeatmap")

for (i in libs.list) {library(i, character.only = TRUE)}

rm(list = c("i", "libs.list"))

dir.create("out/tables")

dir.create("out/plots")
```

```{r , message=FALSE, warning=FALSE, prompt=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# Read data/ Create phyloseq
#-------------------------------------------------------------------------------
ps1 <- qza_to_phyloseq(features = "data/asv_table.qza",
                       tree = "data/rooted-tree.qza",
                       taxonomy = "data/taxonomy_07.qza")

#-------------------------------------------------------------------------------
# Other data
#-------------------------------------------------------------------------------
ps1.meta <- read.csv("data/PERSON_intervention_meta.csv")

rownames(ps1.meta) <- ps1.meta$SeqID

# Pre-selected columns 
col.sel.vec <- c("SeqID", "PartUniq_ID", "ID", "sc_phenotype", 
                  "sc_phenotype_MIRLIR", "CIW", "d_CIW2","center", 
                  "center_name",	"diet",	"diet_optimal",	"diet_name",
                  "sex",	"dropout", "sc_age", "end_med_study.Antibiotics", 
                  "weight", "hfmm_tag_0", "HOMA_IR", "Matsuda", "misi", 
                  "CRP", "log10_misi", "log10_HOMA_IR",	"log10_Matsuda",   
                  "log10_hfmm_tag_0", "log10_CRP")

ps1.meta <- ps1.meta %>% 
                dplyr::select(col.sel.vec) %>% 
                mutate(phenotype_diet = paste0(sc_phenotype_MIRLIR, "_", 
                                               diet_name), 
                       Time_point = paste0("T", CIW))

# Select only relevant columns and add to the phyloseq
sample_data(ps1) <- ps1.meta
```


```{r , message=FALSE, warning=FALSE, prompt=FALSE, results='hide', echo=FALSE}
################################################################################
# Filtering
################################################################################
#-------------------------------------------------------------------------------
# Initial ASV filtering 
#-------------------------------------------------------------------------------
# filter taxa with with less than 10 reads in total   
ps1 <- prune_taxa(taxa_sums(ps1) > 10, ps1)

# Remove ASVs: 
# Kingdom: "d__Eukaryota", "Unassigned"
# Genus: "Mitochondria"
ps1 <- prune_taxa(!tax_table(ps1)[, "Genus"] %in% "Mitochondria", ps1)

ps1 <- prune_taxa(!tax_table(ps1)[, "Kingdom"] %in% c("d__Eukaryota", "Unassigned"), 
                      ps1)

# Change ASV IDs to shortened taxonomic names 
source("R/phy_shorten_tax_names.R")

taxa_names(ps1) <- make.unique(phy_shorten_tax_names(ps1))

# Write taxa table 
tax_table(ps1) %>% 
  as.matrix() %>% 
  as.data.frame() %>% 
  write.csv(., "out/tables/taxa_table.csv")

#-------------------------------------------------------------------------------
# Samples
#-------------------------------------------------------------------------------
# Remove participants that used antibiotics (T0 and T1)
ant.ids <- ps1.meta$ID[ps1.meta$end_med_study.Antibiotics != "0"] 

ps1.f1 <- prune_samples(!ps1.meta$ID %in% ant.ids, ps1)

ps1.f1.meta <- ps1.f1 %>% 
                  sample_data() %>% 
                  as.matrix() %>% 
                  as.data.frame() 


#-------------------------------------------------------------------------------
# List of phyloseq with different normalization
#-------------------------------------------------------------------------------
# Raw - genus 
ps1.f1.genus <- tax_glom(ps1.f1, "Genus")

taxa_names(ps1.f1.genus) <- ps1.f1.genus %>% 
                                phy_shorten_tax_names() %>% 
                                make.unique()

# CSS - ASV
ps1.css.asv <- ps1.f1

otu_table(ps1.css.asv) <- ps1.f1 %>% 
                              phyloseq_to_metagenomeSeq(.) %>% 
                              cumNorm(., p=cumNormStatFast(.)) %>% 
                              MRcounts(., norm=TRUE, log=TRUE) %>% 
                              as.data.frame() %>% 
                              otu_table(., taxa_are_rows = TRUE)


# CSS - Genus
ps1.css.genus <- ps1.f1.genus

otu_table(ps1.css.genus) <- ps1.f1.genus %>% 
                                phyloseq_to_metagenomeSeq(.) %>% 
                                cumNorm(., p=cumNormStatFast(.)) %>% 
                                MRcounts(., norm=TRUE, log=TRUE) %>% 
                                as.data.frame() %>% 
                                otu_table(., taxa_are_rows = TRUE)


# Combine phyloseqs in a list and glom them 
pss.all.ls <-  list(ps.raw.asv = ps1.f1, 
                    ps.raw.genus = ps1.f1.genus,
                    ps.rare.asv = rarefy_even_depth(ps1.f1, 
                                                    rngseed = 4067), 
                    ps.rare.genus = rarefy_even_depth(ps1.f1.genus, 
                                                      rngseed = 4067), 
                    ps.relat.asv = transform_sample_counts(ps1.f1, 
                                                           function(x) x/sum(x)), 
                    ps.relat.genus = transform_sample_counts(ps1.f1.genus, 
                                                           function(x) x/sum(x)),
                    ps.css.asv = ps1.css.asv, 
                    ps.css.genus = ps1.css.genus)
```

# Alpha diversity

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
################################################################################
# Alpha diversity
################################################################################
#-------------------------------------------------------------------------------
# Prepare data and calculate alpha diversity
#-------------------------------------------------------------------------------
# Calculate diversity indexes 
alpha.df <- estimate_richness(physeq = pss.all.ls$ps.rare.asv, 
                  measures = c("Observed", "Shannon", "InvSimpson"))

# Calculate Phylogeny Diversity (package "picante")
alpha.df$PhyloDiverity <- picante::pd(samp = t(otu_table(pss.all.ls$ps.rare.asv)), 
                                       tree = phy_tree(pss.all.ls$ps.rare.asv), 
                                       include.root = FALSE)  %>% 
                          dplyr::select("PD") %>%  
                          unlist()


# Select relevant columns & and convert to long format
alpha.col <- c("sc_phenotype_MIRLIR", "Time_point", "center_name", "ID", "sex",
                                     "diet_name", "sc_age", "phenotype_diet")

alpha.df.long <- ps1.f1.meta %>%
                    dplyr::select(alpha.col) %>% 
                    bind_cols(alpha.df)  %>% 
                    gather(key = "Diversity_Index", 
                           value = "Value", 
                           -all_of(alpha.col)) %>% 
                    mutate(sc_age = as.numeric(sc_age))


#-------------------------------------------------------------------------------
# 1.0. Baseline: T0 between diets with disregard to phenotype.
#-------------------------------------------------------------------------------
# Wilcox test 
# Select only T1 samples
alpha.1.0 <- alpha.df.long[alpha.df.long$Time_point == "T1", ]

wil.res.alpha.1.0 <- NULL

wil.formula <- paste0("Value ~ ", "diet_name")

for (i in unique(alpha.1.0[, "Diversity_Index"])) {
  
  res.wil <- alpha.1.0 %>% 
                filter(Diversity_Index == i) %>% 
                wilcox.test(as.formula(wil.formula), .) %>% 
                tidy() %>% 
                mutate(Index = i)
  
  wil.res.alpha.1.0 <- bind_rows(wil.res.alpha.1.0, res.wil)
  
}


#-------------------------------------------------------------------------------
# 1.1. Pairwise test per subgroup (phenotype_diet) & 
# 1.2. Pairwise test per diet disregarding phenotype (diet) &
# 1.3. Pairwise test per phenotype disregarding diet (phenotype) 
#-------------------------------------------------------------------------------
# Variables 
subcols <- c("phenotype_diet", "diet_name", "sc_phenotype_MIRLIR")

alpha.plots.ls <- list()

alpha.stat.ls <- list()

wil.formula <- paste0("Value ~ ", "Time_point")


# Test in a loop 
for (sub.gr.col in subcols) {
  
  # Stat test
  #-----------
  test.res <- NULL
  
  for (i1 in unique(alpha.df.long[, sub.gr.col])) {
  
    for (i2 in unique(alpha.df.long[, "Diversity_Index"])) {
      
      res.wil <- alpha.df.long %>% 
                    filter(get(sub.gr.col) == i1, 
                           Diversity_Index == i2) %>% 
                    arrange(Time_point, ID) %>% 
                    wilcox.test(as.formula(wil.formula), ., paired = TRUE) %>% 
                    tidy() %>% 
                    mutate(!!sub.gr.col := i1, 
                           Diversity_Index = i2)
      
      test.res <- bind_rows(test.res, res.wil)
    
    }
  
  }
  
  alpha.stat.ls[[sub.gr.col]] <- test.res
  
  
  # Plot data
  #----------
  # Make data frame for significance levels 
  max.div <- alpha.df.long %>% 
                group_by(Diversity_Index, get(sub.gr.col)) %>% 
                slice(which.max(Value)) %>% 
                mutate(y.adj = (Value + Value*0.1), 
                       comp_id = paste0(Diversity_Index, "_", get(sub.gr.col)))
  
  sig.df <- test.res %>% 
              mutate(comp_id = paste0(Diversity_Index, "_", get(sub.gr.col)), 
                     p.short = paste0("p=", round(p.value, 3)), 
                     Start = unique(ps1.f1.meta$Time_point)[1], 
                     End = unique(ps1.f1.meta$Time_point)[2]) %>% 
              filter(p.value <= 0.1) %>% 
              select(-all_of(c("Diversity_Index", sub.gr.col))) %>% 
              left_join(., max.div, by = "comp_id")
  
  
  # Plot alpha diversity 
  alpha.plot <- ggplot(alpha.df.long, aes_string(y = "Value", 
                                                 x = "Time_point")) + 
                    geom_jitter(size = 2, 
                                alpha = 0.5, width = 0.25, height = 0) +
                    geom_boxplot(fill = "black", alpha = 0.1, outlier.colour = NA) +
                    geom_signif(data = sig.df,
                                aes(xmin = Start,
                                    xmax = End,
                                    annotations = p.short,
                                    y_position = y.adj),
                                textsize = 4, vjust = -0.1,
                                manual = TRUE, margin_top = 1) +
                    geom_point(data = sig.df,
                                aes(x = End, y = y.adj*1.1), x=NA) +
                    facet_grid(c("Diversity_Index", sub.gr.col), 
                                      scales = "free") + 
                    theme_bw() + 
                   # scale_color_manual(values = col.gr) + 
                    theme(legend.position = "none")
  
  alpha.plots.ls[[sub.gr.col]] <- alpha.plot

}

#-------------------------------------------------------------------------------
# 1.4 GLM as complete model:
#         "alpha ~ sex + age + place + Phenotype*Diet*Group + (1|Participant)" 
#-------------------------------------------------------------------------------

lmm.res.ls <- list()

for (i in unique(alpha.df.long$Diversity_Index)) {
  
data.lm <- alpha.df.long %>% 
              filter(Diversity_Index == i) %>% 
              mutate_if(is.character, as.factor)

 res.lmer <- lmerTest::lmer(Value ~ sex + 
                              sc_age + 
                              center_name + 
                              sc_phenotype_MIRLIR*diet_name*Time_point + (1|ID), 
                              data = data.lm)
 
 lmm.res.ls[[i]] <- car::Anova(res.lmer)
 
}
```


## Results: Alpha diveristy: Baseline: T0 between diets with disregard to phenotype.

\
```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
kable(wil.res.alpha.1.0[, -3])
```

## Results: Alpha diveristy: Pairwise tests

1.1. Pairwise test per subgroup (phenotype_diet) \
1.2. Pairwise test per diet disregarding phenotype (diet) \
1.3. Pairwise test per phenotype disregarding diet (phenotype) \
\
```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(alpha.stat.ls)) {
  
  print(kable(alpha.stat.ls[[i]][, -3], caption = i ))
  
}
```
\
```{r , message=FALSE, warning=FALSE, echo=FALSE}
alpha.plots.ls
```

## Results: Alpha diveristy: Alpha diversity LMM

Model: Value ~ sex + sc_age + center_name + sc_phenotype_MIRLIR\*diet_name\*Time_point + (1|ID)
\
```{r , message=FALSE, warning=FALSE, echo=FALSE}
lmm.res.ls
```

# Beta diversity

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
################################################################################
# Beta diversity
################################################################################
# Fit ADONIS model to beta diversity ordination
# Normalization: CSS, rarefaction. 
# Level: ASV, Genus. 
#-------------------------------------------------------------------------------

ps.ls <- c("ps.rare.asv", "ps.rare.genus", "ps.relat.asv",
           "ps.relat.genus", "ps.css.asv", "ps.css.genus") 


# List of distances to use 
used.dist <- c("unifrac", "wunifrac", "jaccard", "bray")

#-------------------------------------------------------------------------------
# 2.0.1. Baseline: T0 dist ~ sex + age + place + Phenotype*Diet 
# 2.0.2. Baseline: T1 dist ~ sex + age + place + Phenotype*Diet 
#-------------------------------------------------------------------------------

fact.var <- c("sex", "center_name", "sc_phenotype_MIRLIR", "diet_name")

num.var <- c("sc_age")

# Formula for ADONIS testing 2.0
adonis.formula <- paste0("dist.inst ~ ", "sex", " + ", 
                                         "sc_age", " + ", 
                                         "center_name", " + ", 
                                         "sc_phenotype_MIRLIR", "*", 
                                          "diet_name")

res.adonis.2.0 <- list()

for (i1 in c("T1", "T2")) {
  
  for (i2 in ps.ls) {
    
    ps.dist <- prune_samples(ps1.f1.meta$Time_point == i1, 
                             pss.all.ls[[i2]])
    
    ps.meta <- ps.dist %>% 
                sample_data() %>% 
                as.matrix() %>% 
                as.data.frame() %>% 
                mutate_at(all_of(fact.var), as.factor) %>% 
                mutate_at(all_of(num.var), as.numeric)
      
    
    for (i3 in used.dist) {
        
        res.names <- paste0(i2, "_", i3, "_", i1)  
      
        dist.inst <- distance(ps.dist, method = i3, type = "samples")
  
        res.adonis.2.0[[res.names]] <- adonis2(formula = as.formula(adonis.formula), 
                                                data = ps.meta, 
                                                by = "terms", 
                                                permutations = 9, 
                                                parallel = 4) %>% 
                                          tidy() %>% 
                                          mutate(Distance = i3, 
                                                 Time_point = i1, 
                                                 PS = i2)
    }
    
  }
  
}

#-------------------------------------------------------------------------------
# 2.1.1 Complete model: 
# "dist ~ Participant_ID + sex + age + place + Phenotype*Diet*Group" 
# 2.1.2 Complete model: without random effect 
# "dist ~ sex + age + place + Phenotype*Diet*Group" 
#-------------------------------------------------------------------------------
fact.var <- c("ID", "sex", "center_name", "sc_phenotype_MIRLIR", 
              "diet_name", "Time_point")

num.var <- c("sc_age")

# Formulas for ADONIS testing 2.1
adonis.formula.ls <- list()

adonis.formula.ls[["Random"]]<- paste0("dist.inst ~ ", 
                                        "ID", "+",
                                        "sex", " + ", 
                                        "sc_age", " + ", 
                                        "center_name", " + ", 
                                        "sc_phenotype_MIRLIR", "*", 
                                        "diet_name", "*", 
                                        "Time_point")

adonis.formula.ls[["No_random"]]<- paste0("dist.inst ~ ", 
                                          "sex", " + ", 
                                          "sc_age", " + ", 
                                          "center_name", " + ", 
                                          "sc_phenotype_MIRLIR", "*", 
                                          "diet_name", "*", 
                                          "Time_point")

res.adonis.2.1 <- list()

for (i1 in 1:length(adonis.formula.ls)) {
  
  for (i2 in ps.ls) {
    
    ps.dist <- pss.all.ls[[i2]]
    
    ps.meta <- ps.dist %>% 
                sample_data() %>% 
                as.matrix() %>% 
                as.data.frame() %>% 
                mutate_at(all_of(fact.var), as.factor) %>% 
                mutate_at(all_of(num.var), as.numeric)
    
    for (i3 in used.dist) {
        
        res.names <- paste0(i2, "_", i3, "_", 
                            names(adonis.formula.ls)[i1])  
      
        dist.inst <- distance(ps.dist, method = i3, type = "samples")
  
        res.adonis.2.1[[res.names]] <- adonis2(formula = as.formula(adonis.formula.ls[[i1]]),
                                                data = ps.meta,
                                                by = "terms",
                                                permutations = 9,
                                                parallel = 4) %>%
                                          tidy() %>%
                                          mutate(Distance = i3,
                                                 Random_effect = names(adonis.formula.ls)[i1],
                                                 PS = i2)
    }
    
  }
  
}


#-------------------------------------------------------------------------------
# 2.2.1 Phenotype/diet stratified: 
# "dist ~ Participant_ID + sex + age + place + Group" 
# 2.2.2 Phenotype/diet stratified: \
# "dist ~ sex + age + place + Group" \
#-------------------------------------------------------------------------------

fact.var <- c("ID", "sex", "center_name", "sc_phenotype_MIRLIR", 
              "diet_name", "Time_point")

num.var <- c("sc_age")

# Formulas for ADONIS testing 2.2
adonis.formula.ls <- list()

adonis.formula.ls[["Random"]]<- paste0("dist.inst ~ ", 
                                        "ID", "+",
                                        "sex", " + ", 
                                        "sc_age", " + ", 
                                        "center_name", " + ", 
                                        "Time_point")

adonis.formula.ls[["No_random"]]<- paste0("dist.inst ~ ", 
                                          "sex", " + ", 
                                          "sc_age", " + ", 
                                          "center_name", " + ", 
                                          "Time_point")

res.adonis.2.2 <- list()

for (i1 in unique(ps1.f1.meta$phenotype_diet)) {
  
  for (i2 in ps.ls) {
    
    ps.dist <- prune_samples(ps1.f1.meta$phenotype_diet == i1, 
                             pss.all.ls[[i2]])
    
    ps.meta <- ps.dist %>% 
                sample_data() %>% 
                as.matrix() %>% 
                as.data.frame() %>% 
                mutate_at(all_of(fact.var), as.factor) %>% 
                mutate_at(all_of(num.var), as.numeric)
    
    for (i3 in used.dist) {
      
      for (i4 in 1:length(adonis.formula.ls)) {
        
        res.names <- paste0(i2, "_", i3, "_", 
                            names(adonis.formula.ls)[i4], "_", 
                            i1)  
      
        dist.inst <- distance(ps.dist, method = i3, type = "samples")
  
        res.adonis.2.2[[res.names]] <- adonis2(formula = as.formula(adonis.formula.ls[[i4]]), 
                                                data = ps.meta, 
                                                by = "terms", 
                                                permutations = 9, 
                                                parallel = 4) %>% 
                                          tidy() %>% 
                                          mutate(Distance = i3, 
                                                 Random_effect = names(adonis.formula.ls)[i4], 
                                                 PS = i2, 
                                                 Strata = i1)
      }
      
    }
    
  }
  
}


#-------------------------------------------------------------------------------
# 2.3.1 Diet stratified: 
# "dist ~ Participant_ID + sex + age + place + Group" 
# 2.3.2 Diet stratified: \
# "dist ~ sex + age + place + Group" \
#-------------------------------------------------------------------------------

fact.var <- c("ID", "sex", "center_name", "sc_phenotype_MIRLIR", 
              "diet_name", "Time_point")

num.var <- c("sc_age")

# Formulas for ADONIS testing 2.2
adonis.formula.ls <- list()

adonis.formula.ls[["Random"]]<- paste0("dist.inst ~ ", 
                                        "ID", "+",
                                        "sex", " + ", 
                                        "sc_age", " + ", 
                                        "center_name", " + ", 
                                        "Time_point")

adonis.formula.ls[["No_random"]]<- paste0("dist.inst ~ ", 
                                          "sex", " + ", 
                                          "sc_age", " + ", 
                                          "center_name", " + ", 
                                          "sc_phenotype_MIRLIR",  "*",
                                          "Time_point")

res.adonis.2.3 <- list()

for (i1 in unique(ps1.f1.meta$diet_name)) {
  
  for (i2 in ps.ls) {
    
    ps.dist <- prune_samples(ps1.f1.meta$diet_name == i1, 
                             pss.all.ls[[i2]])
    
    ps.meta <- ps.dist %>% 
                sample_data() %>% 
                as.matrix() %>% 
                as.data.frame() %>% 
                mutate_at(all_of(fact.var), as.factor) %>% 
                mutate_at(all_of(num.var), as.numeric)
    
    for (i3 in used.dist) {
      
      for (i4 in 1:length(adonis.formula.ls)) {
        
        res.names <- paste0(i2, "_", i3, "_", 
                            names(adonis.formula.ls)[i4], "_", 
                            i1)  
      
        dist.inst <- distance(ps.dist, method = i3, type = "samples")
  
        res.adonis.2.3[[res.names]] <- adonis2(formula = as.formula(adonis.formula.ls[[i4]]), 
                                                data = ps.meta, 
                                                by = "terms", 
                                                permutations = 9, 
                                                parallel = 4) %>% 
                                          tidy() %>% 
                                          mutate(Distance = i3, 
                                                 Random_effect = names(adonis.formula.ls)[i4], 
                                                 PS = i2, 
                                                 Strata = i1)
      }
      
    }
    
  }
  
}


#-------------------------------------------------------------------------------
# 2.4.1 Optimal Diet stratified: 
# "dist ~ Participant_ID + sex + age + place + Group" 
# 2.4.2 Optimal Diet stratified: \
# "dist ~ sex + age + place + Group" \
#-------------------------------------------------------------------------------

fact.var <- c("ID", "sex", "center_name", "sc_phenotype_MIRLIR", 
              "diet_name", "Time_point")

num.var <- c("sc_age")

adonis.formula.ls <- list()

adonis.formula.ls[["Random"]]<- paste0("dist.inst ~ ", 
                                        "ID", "+",
                                        "sex", " + ", 
                                        "sc_age", " + ", 
                                        "center_name", " + ", 
                                        "Time_point")

adonis.formula.ls[["No_random"]]<- paste0("dist.inst ~ ", 
                                          "sex", " + ", 
                                          "sc_age", " + ", 
                                          "center_name", " + ", 
                                          "sc_phenotype_MIRLIR",  "*",
                                          "Time_point")

res.adonis.2.4 <- list()

for (i1 in unique(ps1.f1.meta$diet_optimal)) {
  
  for (i2 in ps.ls) {
    
    ps.dist <- prune_samples(ps1.f1.meta$diet_optimal == i1, 
                             pss.all.ls[[i2]])
    
    ps.meta <- ps.dist %>% 
                sample_data() %>% 
                as.matrix() %>% 
                as.data.frame() %>% 
                mutate_at(all_of(fact.var), as.factor) %>% 
                mutate_at(all_of(num.var), as.numeric)
    
    for (i3 in used.dist) {
      
      for (i4 in 1:length(adonis.formula.ls)) {
        
        res.names <- paste0(i2, "_", i3, "_", 
                            names(adonis.formula.ls)[i4], "_", 
                            i1)  
      
        dist.inst <- distance(ps.dist, method = i3, type = "samples")
  
        res.adonis.2.4[[res.names]] <- adonis2(formula = as.formula(adonis.formula.ls[[i4]]), 
                                                data = ps.meta, 
                                                by = "terms", 
                                                permutations = 9, 
                                                parallel = 4) %>% 
                                          tidy() %>% 
                                          mutate(Distance = i3, 
                                                 Random_effect = names(adonis.formula.ls)[i4], 
                                                 PS = i2, 
                                                 Strata = i1)
      }
      
    }
    
  }
  
}

```

## Results: Beta diversity: Baseline

2.0.1. Baseline: T0 dist ~ sex + age + place + Phenotype*Diet \
2.0.2. Baseline: T1 dist ~ sex + age + place + Phenotype*Diet \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(res.adonis.2.0)) {
  
  print(kable(res.adonis.2.0[[i]], caption = i ))
  
}
```

## Results: Beta diversity: Complete model

2.1.1 Complete model: \
"dist ~ Participant_ID + sex + age + place + Phenotype\*Diet\*Group" \
2.1.2 Complete model: without random effect \
"dist ~ sex + age + place + Phenotype\*Diet\*Group" \
\
```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(res.adonis.2.1)) {
  
  print(kable(res.adonis.2.1[[i]], caption = i ))
  
}
```

## Results: Beta diversity: Phenotype/diet stratified

2.2.1 Phenotype/diet stratified: \
"dist ~ Participant_ID + sex + age + place + Group" \
2.2.2 Phenotype/diet stratified: \
"dist ~ sex + age + place + Group" \
\
```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(res.adonis.2.2)) {
  
  print(kable(res.adonis.2.2[[i]], caption = i ))
  
}
```

## Results: Beta diversity: Diet stratified

2.3.1 Diet stratified: \
"dist ~ Participant_ID + sex + age + place + Group" \
2.3.2 Diet stratified: \
"dist ~ sex + age + place + Group" \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(res.adonis.2.3)) {
  
  print(kable(res.adonis.2.3[[i]], caption = i ))
  
}
```

## Results: Beta diversity: Optimal Diet stratified

2.4.1 Optimal Diet stratified: \
"dist ~ Participant_ID + sex + age + place + Group" \
2.4.2 Optimal Diet stratified: \
"dist ~ sex + age + place + Group" \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
for (i in names(res.adonis.2.4)) {
  
  print(kable(res.adonis.2.4[[i]], caption = i ))
  
}
```

# DA of taxa between T0 and T1

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# MAASLIN: Extract data
#-------------------------------------------------------------------------------
source("R/phy_taxa_filter.R")

ps.ls.maas <- c("ps.raw.asv", "ps.raw.genus")
  
out.folder <- "out/maaslin_final/"

dir.create(out.folder, recursive = TRUE)


#-------------------------------------------------------------------------------
# Stratification by phenotype & diet 
#-------------------------------------------------------------------------------
res.maas.long.1 <- NULL

for (i1 in ps.ls.maas) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta$phenotype_diet))  {
  
    ps.gr <- prune_samples(ps1.f1.meta$phenotype_diet == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "Time_point")
    
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      as.data.frame() 
       
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame()
    
       maas.out <- Maaslin2(
                          input_data =  ps.gr.asv, 
                          input_metadata = ps.gr.meta, 
                          output = paste0(out.folder, inst.name), 
                          fixed_effects = c("Time_point"), 
                          random_effects = "ID", 
                          reference = "Time_point,T1", 
                          correction = "BH", 
                          cores = 4, 
                          min_abundance = 0, 
                          min_prevalence = 0, 
                          min_variance = 0, 
                          normalization = "CSS", 
                          transform = "LOG", 
                          analysis_method = "LM", 
                          max_significance = 0.25)
    
  
  res.maas.long.1 <- rbind(res.maas.long.1, cbind(maas.out$results, 
                                              Strata = inst.name, 
                                              Tax_level = i1, 
                                              phenotype_diet = i2, 
                                              Cut_off = i3))  

      }
 
    }
  
}

# Plot Maaslin results 
#---------------------

res.maas.long.f1 <- res.maas.long.1 %>% 
                    filter(qval <= 0.2, 
                           Cut_off > 0.2)

maas.plot.1 <- ggplot(res.maas.long.f1, 
                        aes(x = coef, y = feature)) + 
                        geom_segment(aes(x = 0, xend = coef, 
                                     y = feature, yend = feature), 
                                     color = "black", alpha = 0.75) + 
                        geom_vline(xintercept = 0, alpha = 0.75) + 
                        geom_point(size = 3.5, alpha = 0.9) + 
                        facet_wrap(. ~ Strata, scales = "free_y", ncol = 2) + 
                        theme_bw() +
                        theme(panel.grid.major.x = element_blank(),
                              panel.grid.minor.x = element_blank(), 
                              axis.text.y = element_text(face = "italic")) +
                        ylab("") + xlab("Coeficient")


#-------------------------------------------------------------------------------
# Stratification by diet ONLY
#-------------------------------------------------------------------------------
res.maas.long.2 <- NULL

for (i1 in ps.ls.maas) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta$diet_name))  {
  
    ps.gr <- prune_samples(ps1.f1.meta$diet_name == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "diet_name")
    
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      as.data.frame() 
       
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame()
    
       maas.out <- Maaslin2(
                          input_data =  ps.gr.asv, 
                          input_metadata = ps.gr.meta, 
                          output = paste0(out.folder, inst.name), 
                          fixed_effects = c("Time_point"), 
                          random_effects = "ID", 
                          reference = "Time_point,T1", 
                          correction = "BH", 
                          cores = 4, 
                          min_abundance = 0, 
                          min_prevalence = 0, 
                          min_variance = 0, 
                          normalization = "CSS", 
                          transform = "LOG", 
                          analysis_method = "LM", 
                          max_significance = 0.25)
    
  
  res.maas.long.2 <- rbind(res.maas.long.2, cbind(maas.out$results, 
                                              Strata = inst.name, 
                                              Tax_level = i1, 
                                              phenotype_diet = i2, 
                                              Cut_off = i3))  

      }
 
    }
  
}


# Plot Maaslin results 
#---------------------
res.maas.long.f2 <- res.maas.long.2 %>% 
                    filter(qval <= 0.2, 
                           Cut_off > 0.2)

maas.plot.2 <- ggplot(res.maas.long.f2, 
                        aes(x = coef, y = feature)) + 
                        geom_segment(aes(x = 0, xend = coef, 
                                     y = feature, yend = feature), 
                                     color = "black", alpha = 0.75) + 
                        geom_vline(xintercept = 0, alpha = 0.75) + 
                        geom_point(size = 3.5, alpha = 0.9) + 
                        facet_wrap(. ~ Strata, scales = "free_y", ncol = 2) + 
                        theme_bw() +
                        theme(panel.grid.major.x = element_blank(),
                              panel.grid.minor.x = element_blank(), 
                              axis.text.y = element_text(face = "italic")) +
                        ylab("") + xlab("Coeficient")


#-------------------------------------------------------------------------------
# Stratification by diet: Optimal vs Suboptimal
#-------------------------------------------------------------------------------
res.maas.long.3 <- NULL

for (i1 in ps.ls.maas) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta$diet_optimal))  {
  
    ps.gr <- prune_samples(ps1.f1.meta$diet_optimal == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "diet_optimal")
    
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      as.data.frame() 
       
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame()
    
       maas.out <- Maaslin2(
                          input_data =  ps.gr.asv, 
                          input_metadata = ps.gr.meta, 
                          output = paste0(out.folder, inst.name), 
                          fixed_effects = c("Time_point"), 
                          random_effects = "ID", 
                          reference = "Time_point,T1", 
                          correction = "BH", 
                          cores = 4, 
                          min_abundance = 0, 
                          min_prevalence = 0, 
                          min_variance = 0, 
                          normalization = "CSS", 
                          transform = "LOG", 
                          analysis_method = "LM", 
                          max_significance = 0.25)
    
  
  res.maas.long.3 <- rbind(res.maas.long.3, cbind(maas.out$results, 
                                              Strata = inst.name, 
                                              Tax_level = i1, 
                                              phenotype_diet = i2, 
                                              Cut_off = i3))  

      }
 
    }
  
}


# Plot Maaslin results 
#---------------------
res.maas.long.f3 <- res.maas.long.3 %>% 
                    filter(qval <= 0.2, 
                           Cut_off > 0.2)

maas.plot.3 <- ggplot(res.maas.long.f3, 
                        aes(x = coef, y = feature)) + 
                        geom_segment(aes(x = 0, xend = coef, 
                                     y = feature, yend = feature), 
                                     color = "black", alpha = 0.75) + 
                        geom_vline(xintercept = 0, alpha = 0.75) + 
                        geom_point(size = 3.5, alpha = 0.9) + 
                        facet_wrap(. ~ Strata, scales = "free_y") + 
                        theme_bw() +
                        theme(panel.grid.major.x = element_blank(),
                              panel.grid.minor.x = element_blank(), 
                              axis.text.y = element_text(face = "italic")) +
                        ylab("") + xlab("Coeficient")
```

## Results: DA: MaAsLin2

### Results: DA: MaAsLin2: Phenotype/diet stratified

**Phenotype/diet stratified:** (MaAsLin) term - Group, random effect - Participant \
\
**Table:** Fifty most significantly different taxa according to Maaslin2. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
maap.print.tab <- res.maas.long.1[order(res.maas.long.1$qval), ] %>% 
                    .[1:50, ]

print(kable(maap.print.tab))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Maaslin2. Minimal taxa prevalence per within group 50%. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=8, fig.height=8}
maas.plot.1
```

### Results: DA: MaAsLin2: Diet stratified

**Table:** Fifty most significantly different taxa according to Maaslin2. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
maap.print.tab <- res.maas.long.2[order(res.maas.long.2$qval), ] %>% 
                    .[1:50, ]

print(kable(maap.print.tab))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Maaslin2. Minimal taxa prevalence per within group 50%. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=8, fig.height=10}
maas.plot.2
```

### Results: DA: MaAsLin2: Diet (optimal/suboptimal) stratified

**Table:** Fifty most significantly different taxa according to Maaslin2. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
maap.print.tab <- res.maas.long.3[order(res.maas.long.3$qval), ] %>% 
                    .[1:50, ]

print(kable(maap.print.tab))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Maaslin2. Minimal taxa prevalence per within group 50%. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=8, fig.height=7}
maas.plot.3
```

## Results: DA: Wilcox (paired)

Paired Wilcox by Group. \
Normalization: CSS, Rarefaction & relative abundance \

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# Wilcoxon - paired test
#-------------------------------------------------------------------------------
source("R/phy_taxa_filter.R")

ps.ls.will <- c("ps.rare.asv", "ps.rare.genus", "ps.relat.asv", 
                "ps.relat.genus", "ps.css.asv", "ps.css.genus")

#-------------------------------------------------------------------------------
# Stratified by diet_phenotype
#-------------------------------------------------------------------------------

strata.col <- "phenotype_diet"

res.wilcox.long.1 <- NULL

for (i1 in ps.ls.will) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta[, strata.col]))  {
  
    ps.gr <- prune_samples(ps1.f1.meta[, strata.col] == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "Time_point")
    
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame() %>% 
                      group_by(ID) %>% 
                      arrange(Time_point, .by_group = TRUE)
       
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      t() %>% 
                      as.data.frame() 
       
       ps.gr.asv <- ps.gr.asv[ps.gr.meta$SeqID, ] %>% 
                      mutate(Time_point = ps.gr.meta$Time_point)
       
       res.tax.wil <- NULL
       
       for(i4 in colnames(ps.gr.asv)) {
         
         if(i4 != "Time_point") {
           
          res.tax.wil <- wilcox.test(as.formula(paste0(i4, "~", "Time_point")), 
                     paired = TRUE, data = ps.gr.asv) %>% 
                     tidy() %>% 
                     mutate(taxa = i4, 
                            PS = i1, 
                            Strata = i2, 
                            Cut_off = i3) %>% 
                     bind_rows(res.tax.wil, .)
           
         }
         
       }
    
  
       res.tax.wil$qval <- p.adjust(res.tax.wil$p.value, method = "BH")
       
       res.wilcox.long.1 <- bind_rows(res.wilcox.long.1, res.tax.wil)
    
      }
 
    }
  
}

# Plot Wilcox data 
#-----------------
res.wil.p <- res.wilcox.long.1[res.wilcox.long.1$qval <= 0.2, ] %>% 
                      mutate(Comb_meta = paste0(Strata, "--", 
                                                PS, "--", 
                                                Cut_off)) %>% 
                      select(c("taxa", "qval", "Comb_meta")) %>% 
                      spread(Comb_meta, qval) %>% 
                      column_to_rownames("taxa")
    
res.wil.p[is.na(res.wil.p)] <- 1 

wil.heat.genus.1 <- res.wil.p[, grepl("genus", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "Genus")

wil.heat.asv.1 <- res.wil.p[, grepl("asv", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "ASV")


#-------------------------------------------------------------------------------
# Stratified by diet 
#-------------------------------------------------------------------------------

strata.col <- "diet_name"

res.wilcox.long.2 <- NULL

for (i1 in ps.ls.will) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta[, strata.col]))  {
  
    ps.gr <- prune_samples(ps1.f1.meta[, strata.col] == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "Time_point")
    
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame() %>% 
                      group_by(ID) %>% 
                      arrange(Time_point, .by_group = TRUE)
       
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      t() %>% 
                      as.data.frame() 
       
       ps.gr.asv <- ps.gr.asv[ps.gr.meta$SeqID, ] %>% 
                      mutate(Time_point = ps.gr.meta$Time_point)
       
       res.tax.wil <- NULL
       
       for(i4 in colnames(ps.gr.asv)) {
         
         if(i4 != "Time_point") {
           
          res.tax.wil <- wilcox.test(as.formula(paste0(i4, "~", "Time_point")), 
                     paired = TRUE, data = ps.gr.asv) %>% 
                     tidy() %>% 
                     mutate(taxa = i4, 
                            PS = i1, 
                            Strata = i2, 
                            Cut_off = i3) %>% 
                     bind_rows(res.tax.wil, .)
           
         }
         
       }
    
  
       res.tax.wil$qval <- p.adjust(res.tax.wil$p.value, method = "BH")
       
       res.wilcox.long.2 <- bind_rows(res.wilcox.long.2, res.tax.wil)
    
      }
 
    }
  
}

# Plot Wilcox data 
#-----------------
res.wil.p <- res.wilcox.long.2[res.wilcox.long.2$qval <= 0.2, ] %>% 
                      mutate(Comb_meta = paste0(Strata, "--", 
                                                PS, "--", 
                                                Cut_off)) %>% 
                      select(c("taxa", "qval", "Comb_meta")) %>% 
                      spread(Comb_meta, qval) %>% 
                      column_to_rownames("taxa")
    
res.wil.p[is.na(res.wil.p)] <- 1 

wil.heat.genus.2 <- res.wil.p[, grepl("genus", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "Genus")

wil.heat.asv.2 <- res.wil.p[, grepl("asv", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "ASV")

#-------------------------------------------------------------------------------
# Stratified by diet optimal/suboptimal
#-------------------------------------------------------------------------------

strata.col <- "diet_optimal"

res.wilcox.long.3 <- NULL

for (i1 in ps.ls.will) {
  
  ps <- pss.all.ls[[i1]]
  
  for (i2 in unique(ps1.f1.meta[, strata.col]))  {
  
    ps.gr <- prune_samples(ps1.f1.meta[, strata.col] == i2, ps)
  
    for (i3 in c(0.1, 0.25, 0.5)) {
      
       inst.name <- paste0(i1, "_", i2, "_", i3)
      
       ps.gr <- phy_taxa_filter(ps.gr, 
                                prev = i3, 
                                group_col = "Time_point")
    
       ps.gr.meta <- ps.gr %>% 
                      sample_data() %>% 
                      as.matrix() %>% 
                      as.data.frame() %>% 
                      group_by(ID) %>% 
                      arrange(Time_point, .by_group = TRUE)
       
       ps.gr.asv <- ps.gr %>% 
                      otu_table()  %>% 
                      as.matrix()  %>% 
                      t() %>% 
                      as.data.frame() 
       
       ps.gr.asv <- ps.gr.asv[ps.gr.meta$SeqID, ] %>% 
                      mutate(Time_point = ps.gr.meta$Time_point)
       
       res.tax.wil <- NULL
       
       for(i4 in colnames(ps.gr.asv)) {
         
         if(i4 != "Time_point") {
           
          res.tax.wil <- wilcox.test(as.formula(paste0(i4, "~", "Time_point")), 
                     paired = TRUE, data = ps.gr.asv) %>% 
                     tidy() %>% 
                     mutate(taxa = i4, 
                            PS = i1, 
                            Strata = i2, 
                            Cut_off = i3) %>% 
                     bind_rows(res.tax.wil, .)
           
         }
         
       }
    
  
       res.tax.wil$qval <- p.adjust(res.tax.wil$p.value, method = "BH")
       
       res.wilcox.long.3 <- bind_rows(res.wilcox.long.3, res.tax.wil)
    
      }
 
    }
  
}

# Plot Wilcox data 
#-----------------
res.wil.p <- res.wilcox.long.3[res.wilcox.long.3$qval <= 0.2, ] %>% 
                      mutate(Comb_meta = paste0(Strata, "--", 
                                                PS, "--", 
                                                Cut_off)) %>% 
                      select(c("taxa", "qval", "Comb_meta")) %>% 
                      spread(Comb_meta, qval) %>% 
                      column_to_rownames("taxa")
    
res.wil.p[is.na(res.wil.p)] <- 1 


wil.heat.genus.3 <- res.wil.p[, grepl("genus", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "Genus")

wil.heat.asv.3 <- res.wil.p[, grepl("asv", colnames(res.wil.p))] %>% 
                    .[rowSums(.) != ncol(.), ] %>% 
                    Heatmap(., 
                            column_split = gsub("--.*", "", colnames(.)), 
                            row_dend_width = unit(1, "in"),
                            rect_gp = gpar(col = "gray30", lwd = 0.01), 
                            column_names_rot = 45, 
                            name = "q-value", 
                            row_title = "ASV")


```

### Results: DA: Wilcox (paired): Diet/Phenotype

**Table:** Fifty most significantly different taxa according to Wilcox test. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.wilcox.print <- res.wilcox.long.1[order(res.wilcox.long.1$qval), ] %>% 
  .[1:50, -3]

print(kable(res.wilcox.print))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **ASV level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=7, fig.height=4}
wil.heat.asv.1
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **Genus level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=8, fig.height=7}
wil.heat.genus.1
```

### Results: DA: Wilcox (paired): Diet

**Table:** Fifty most significantly different taxa according to Wilcox test. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.wilcox.print <- res.wilcox.long.2[order(res.wilcox.long.2$qval), ] %>% 
  .[1:50, -3]

print(kable(res.wilcox.print))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **ASV level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=9, fig.height=12}
wil.heat.asv.2
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **Genus level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=9, fig.height=12}
wil.heat.genus.2
```


### Results: DA: Wilcox (paired): Diet - Optimal/Suboptimal

**Table:** Fifty most significantly different taxa according to Wilcox test. \

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.wilcox.print <- res.wilcox.long.3[order(res.wilcox.long.3$qval), ] %>% 
  .[1:50, -3]

print(kable(res.wilcox.print))
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **ASV level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=8, fig.height=9}
wil.heat.asv.3
```
\
\
\
**Figure:** Significantly different taxa (q <= 0.2) between groups according to 
Wilcoxon test. **Genus level**\

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=9, fig.height=10}
wil.heat.genus.3
```

## Results: DA: Visualization - box

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}

################################################################################
# Function to plot box plots 
################################################################################
box_plot_taxa <- function(taxa_to_plot, 
                          phylos_list, 
                          plot_group_col,
                          facet_ncol = 4,
                          facet_group_col = NULL,
                          paired_group_col = NULL) {
  
  meta_data_df <- phylos_list[[1]] %>% 
                    sample_data() %>% 
                    as.matrix() %>% 
                    as.data.frame()
  
  # Vector of columns 
  take.col.vec <- c(facet_group_col, 
                    paired_group_col, 
                    plot_group_col)
  
  comb.long.data <- NULL
    
  # Select relevant taxa from otu tables 
  for (i.df in names(phylos_list)) {
    
    i.otu.df <- phylos_list[[i.df]] %>% 
                  otu_table() %>% 
                  as.matrix() %>% 
                  as.data.frame()
    
    if(!all(rownames(i.otu.df) == rownames(meta_data_df))) {
      
      i.otu.df <- t(i.otu.df) %>% 
                    as.data.frame()
      
    }
    
    comb.long.data <- i.otu.df %>% 
                      select(taxa_to_plot) %>% 
                      mutate(Rows_IDs_plot = rownames(.), 
                             OTU_tab_ID = i.df) %>% 
                      bind_cols(., select(meta_data_df, take.col.vec)) %>% 
                      pivot_longer(cols = taxa_to_plot,
                                   names_to = "Taxa_IDs",
                                   values_to = "Values_to_plot") %>% 
                      group_by_at(c("OTU_tab_ID", "Taxa_IDs", 
                                    facet_group_col, paired_group_col)) %>% 
                      arrange(plot_group_col, .by_group = TRUE) %>% 
                      mutate(delta_change = ifelse(Values_to_plot[1] > Values_to_plot[2], 
                                            "Decrease", "Increase")) %>% 
                      mutate(delta_change = ifelse(Values_to_plot[1] == Values_to_plot[2], 
                                            "No_Change", delta_change)) %>% 
                      rbind(comb.long.data, .)
    
  }
  
  
  plot.out <- ggplot(comb.long.data, 
                aes_string(x = plot_group_col, y = "Values_to_plot")) + 
                geom_boxplot() + 
                facet_wrap(c("Taxa_IDs", facet_group_col, "OTU_tab_ID"), 
                           scales = "free", 
                           ncol = facet_ncol) + 
                theme_bw() + 
                ylab(label = "value")
    
  if(!is.null(paired_group_col)) {
    
    plot.out <- plot.out + 
                  geom_line(aes_string(group = paired_group_col, 
                                                 color = "delta_change"), 
                                      alpha = 0.5) + 
                  scale_color_manual(values = c(Increase = "forestgreen", 
                                                Decrease = "firebrick4", 
                                                No_Change = "grey50"))
    
  }
  
  return(plot.out)
              
}
################################################################################

plot.ps.asv <- c("ps.relat.asv", "ps.rare.asv", "ps.css.asv")

plot.ps.genus <- c("ps.relat.genus", "ps.rare.genus", "ps.css.genus")
```

### Results: DA: Visualization: Box plot: Wilcoxon

#### Results: DA: Visualization: Box plot - ASV, wilcoxon, phenotype/diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=20, fig.height=5}
tax.plot.vec <- res.wilcox.long.1 %>% 
                    filter(grepl("asv", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 12, 
              plot_group_col = "Time_point", 
              facet_group_col = "phenotype_diet", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - Genus, wilcoxon, phenotype/diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=24, fig.height=57.5}
tax.plot.vec <- res.wilcox.long.1 %>% 
                    filter(grepl("genus", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 12, 
              plot_group_col = "Time_point", 
              facet_group_col = "phenotype_diet", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - ASV, wilcoxon, diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=100}
tax.plot.vec <- res.wilcox.long.2 %>% 
                    filter(grepl("asv", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_name", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - Genus, wilcoxon, diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=102}
tax.plot.vec <- res.wilcox.long.2 %>% 
                    filter(grepl("genus", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_name", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - ASV, wilcoxon, diet optimal/suboptimal

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=66}
tax.plot.vec <- res.wilcox.long.3 %>% 
                    filter(grepl("asv", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_optimal", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - Genus, wilcoxon, diet optimal/suboptimal

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=74}
tax.plot.vec <- res.wilcox.long.3 %>% 
                    filter(grepl("genus", PS), qval <= 0.2) %>% 
                    pull(taxa) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_optimal", 
              paired_group_col = "ID")

```

### Results: DA: Visualization: Box plot: MaAsLin2

#### Results: DA: Visualization: Box plot - ASV, Maaslin, phenotype/diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=20, fig.height=18}
tax.plot.vec <- res.maas.long.1 %>% 
                    filter(grepl("asv", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 12, 
              plot_group_col = "Time_point", 
              facet_group_col = "phenotype_diet", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - Genus, Maaslin, phenotype/diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=20, fig.height=10}
tax.plot.vec <- res.maas.long.1 %>% 
                    filter(grepl("genus", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 12, 
              plot_group_col = "Time_point", 
              facet_group_col = "phenotype_diet", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - ASV, Maaslin, diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=34}
tax.plot.vec <- res.maas.long.2 %>% 
                    filter(grepl("asv", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_name", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - genus, Maaslin, diet

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=34}
tax.plot.vec <- res.maas.long.2 %>% 
                    filter(grepl("genus", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_name", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - asv, Maaslin, diet optimal/suboptimal

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=26}
tax.plot.vec <- res.maas.long.3 %>% 
                    filter(grepl("asv", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.asv], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_optimal", 
              paired_group_col = "ID")

```

#### Results: DA: Visualization: Box plot - genus, Maaslin, diet optimal/suboptimal

```{r , message=FALSE, warning=FALSE, echo=FALSE, fig.width=15, fig.height=42}
tax.plot.vec <- res.maas.long.3 %>% 
                    filter(grepl("genus", Tax_level), qval <= 0.2) %>% 
                    pull(feature) %>% 
                    unique()

box_plot_taxa(taxa_to_plot = tax.plot.vec, 
              phylos_list = pss.all.ls[plot.ps.genus], 
              facet_ncol = 6, 
              plot_group_col = "Time_point", 
              facet_group_col = "diet_optimal", 
              paired_group_col = "ID")

```

## DA: Results: Summary table 

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
sig.f.all <- c(paste0(rep("res.maas.long.", 3), 1:3), 
                  paste0(rep("res.wilcox.long.", 3), 1:3))

#-------------------------------------------------------------------------------
# Extract significant taxa 
#-------------------------------------------------------------------------------
comb.res.da <- NULL

for(i.sig.f in sig.f.all) {
  
  if(grepl("maas", i.sig.f)) {
    
   i.sig.obj <- get(i.sig.f) %>% 
                    select(-Strata) %>% 
                    rename(taxa = feature,  
                           PS = Tax_level, 
                           Strata = phenotype_diet, 
                           p.value = pval)
   
  } else {i.sig.obj <- get(i.sig.f)}
  
  comb.res.da <- i.sig.obj %>% 
                     select(c("taxa", "PS", "Strata", "p.value", "qval", "Cut_off")) %>% 
                     mutate(Test = gsub("res.|.long..", "", i.sig.f)) %>% 
                     bind_rows(comb.res.da, .)
}


comb.res.da.f.asv <- comb.res.da %>% 
                        filter(qval <= 0.2, grepl("asv", PS))

res.da.sum.asv <- NULL

for(i.tax in unique(comb.res.da.f.asv$taxa)) {
  
  i.tax.sub <- comb.res.da.f.asv[comb.res.da.f.asv$taxa == i.tax, ]
  
   res.da.sum.asv <- c(Taxa = i.tax,
     Normalization = paste(gsub("ps.|.asv", "", unique(i.tax.sub$PS)), collapse = "; "), 
     p.value = paste(round(i.tax.sub$p.value, 3), collapse = "; "), 
     q.value = paste(round(i.tax.sub$qval, 3), collapse = "; "), 
     Sig.group = paste(unique(i.tax.sub$Strata), collapse = "; "), 
     Test = paste(unique(i.tax.sub$Test), collapse = "; "), 
     Cut_off = paste(unique(i.tax.sub$Cut_off), collapse = "; ")) %>% 
     rbind(res.da.sum.asv, .)
  
}

res.da.sum.asv <- as.data.frame(res.da.sum.asv)

write.csv(res.da.sum.asv, "out/tables/DA_summary_asv.csv")


#-------------------------------------------------------------------------------
# Genus level
comb.res.da.f.genus <- comb.res.da %>% 
                        filter(qval <= 0.2, grepl("genus", PS))

res.da.sum.genus <- NULL

for(i.tax in unique(comb.res.da.f.genus$taxa)) {
  
  i.tax.sub <- comb.res.da.f.genus[comb.res.da.f.genus$taxa == i.tax, ]
  
   res.da.sum.genus <- c(Taxa = i.tax,
     Normalization = paste(gsub("ps.|.genus", "", unique(i.tax.sub$PS)), collapse = "; "), 
     p.value = paste(round(i.tax.sub$p.value, 3), collapse = "; "), 
     q.value = paste(round(i.tax.sub$qval, 3), collapse = "; "), 
     Sig.group = paste(unique(i.tax.sub$Strata), collapse = "; "), 
     Test = paste(unique(i.tax.sub$Test), collapse = "; "), 
     Cut_off = paste(unique(i.tax.sub$Cut_off), collapse = "; ")) %>% 
     rbind(res.da.sum.genus, .)
  
}

res.da.sum.genus <- as.data.frame(res.da.sum.genus)

write.csv(res.da.sum.genus, "out/tables/DA_summary_genus.csv")

```

# Correlation microbiome metabolic parameters.

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}

phy_corr_matrix <- function(phyloseqs_named_list, 
                            second_df, 
                            cut_offs_vector = 0.5, 
                            by_group = NULL, 
                            corr_method = "spearman") {
  
  # Extract metadata
  ps.meta <- phyloseqs_named_list[[1]] %>% 
                          sample_data() %>% 
                          as.matrix() %>% 
                          as.data.frame()
  
  # Sort second data frame by phyloseq names 
  second.df <- second_df[rownames(ps.meta), ]
  
  if(!identical(rownames(ps.meta), rownames(second.df))) {
    stop("Rows in phyloseq and second_df are not identical")}
  
  
  if (!is.null(by_group)) {
      par.set <- crossing(names(phyloseqs_named_list), 
                  cut_offs_vector, 
                  unique(ps.meta[, by_group]))
  
  } else { par.set <- crossing(names(phyloseqs_named_list), 
                       cut_offs_vector)}

  cor.res.all <- NULL 
  
  for (i1.par in 1:nrow(par.set)) {
    
    par.inst <- par.set[i1.par, ] %>% 
                  as.character()
    
    ps.inst <- phyloseqs_named_list[[par.inst[1]]] 
    
    
    if(!is.null(by_group)) { 
      
        ps.inst.f <- prune_samples(ps.meta[, by_group] == par.inst[3], ps.inst) %>% 
                          phy_taxa_filter(., prev_fraction = as.numeric(par.inst[2])) 
        
        second.df.f <- second.df[ps.meta[, by_group] == par.inst[3], ]
        
    } else {
    
        ps.inst.f <- phy_taxa_filter(ps.inst, 
                                   prev_fraction = as.numeric(par.inst[2])) 
        
        second.df.f <- second.df
          
        }
    
    ps.inst.otu <- ps.inst.f %>% 
                      otu_table() %>% 
                      as.matrix() %>% 
                      as.data.frame()
    
    if(taxa_are_rows(ps.inst.f)) {
      
      ps.inst.otu <- ps.inst.otu %>% 
                        t() %>% 
                        as.data.frame()
      
    }
    
    for (i2.second in colnames(second.df.f)) {
      
      cor.res <- NULL
      
      for (i3.tax in colnames(ps.inst.otu)) {
        
        cor.res <- cor.test(second.df.f[, i2.second],
                            ps.inst.otu[, i3.tax],
                            method = "spearman") %>%
                      tidy() %>%
                      mutate(Cor_vec = i2.second,
                             Taxa = i3.tax,
                             Cut_Off = par.inst[2],
                             Phyloseq = par.inst[1], 
                             Strata = par.inst[3]) %>%
                       bind_rows(cor.res, .)
      }
      
      cor.res$qvalue <- cor.res$p.value %>% 
                            p.adjust(., method = "fdr") 
      
      cor.res.all <- bind_rows(cor.res.all, cor.res)
      
    }
    
  } # End of parameters loop 
  
  return(cor.res.all)
  
} # Function end 


#-------------------------------------------------------------------------------
# Prepare metabolite data 
#-------------------------------------------------------------------------------
# Select column for calculation of delta
delt.cols <- c("Matsuda", "misi", "HOMA_IR", "CRP",
                "log10_Matsuda", "log10_misi", "log10_HOMA_IR", "log10_CRP")

ps1.f1.meta.delta <- ps1.f1.meta %>%
                        group_by(ID) %>%
                        arrange(Time_point, .by_group = TRUE) %>%
                        select(delt.cols) %>%
                        mutate_at(vars(-group_cols(), delt.cols),
                          function(x) {as.numeric(x[2]) - as.numeric(x[1])}) %>%
                        ungroup() %>%
                        select(-ID) %>%
                        setNames(., paste0("delta_", colnames(.))) %>%
                        cbind(ps1.f1.meta, .)

delta.metabol.ciw1 <-  ps1.f1.meta.delta %>%
                            filter(CIW == "1") %>%
                            select_if(grepl("delta_", colnames(ps1.f1.meta.delta)))
```

```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# T1 and delta parameters correlation 
#-------------------------------------------------------------------------------
# Prepare phyloseqs
pss.cor <- pss.all.ls[4:8]

for (i.ps in names(pss.cor)) {
  
  pss.cor[[i.ps]] <- pss.cor[[i.ps]] %>% 
                     prune_samples(sample_data(pss.cor[[i.ps]])$Time_point == "T1", .)
  
}

# Run correlations 
#-------------------------------------------------------------------------------

strata.vec <- c("phenotype_diet", "diet_name", "diet_optimal", "all")

corr.res.T1 <- NULL
  
for (i.strat in strata.vec) {
  
  if(i.strat == "all") {i.strat <- NULL}
  
  corr.res.T1 <- phy_corr_matrix(phyloseqs_named_list = pss.cor, 
                                  second_df = delta.metabol.ciw1, 
                                  cut_offs_vector = c(0.5, 0.75), 
                                  by_group = i.strat, 
                                  corr_method = "spearman") %>% 
                    mutate(Strata2 = i.strat) %>% 
                    bind_rows(corr.res.T1, .)
  
}

write.csv(corr.res.T1, "out/tables/corr_res_T1_delta.csv")
```


```{r , message=FALSE, warning=FALSE, results='hide', echo=FALSE}
#-------------------------------------------------------------------------------
# delta taxa and delta parameters correlation 
#-------------------------------------------------------------------------------
# Prepare phyloseqs
pss.cor.delta <- pss.all.ls[4:8]

for (i.ps in names(pss.cor.delta)) {
  
  ps.inst <- pss.cor.delta[[i.ps]] %>% 
                phy_taxa_filter(., prev_fraction = 0.25)
  
  otu.tab.delta <-  ps.inst %>% 
                      otu_table() %>% 
                      as.matrix() %>% 
                      t() %>% 
                      as.data.frame() %>% 
                      mutate(Time_point = sample_data(ps.inst)$Time_point, 
                             ID = sample_data(ps.inst)$ID, 
                             SeqID = sample_data(ps.inst)$SeqID)    %>% 
                        group_by(ID) %>%
                        arrange(Time_point, .by_group = TRUE) %>%
                        mutate_at(vars(-group_cols(), -Time_point, -SeqID),
                          function(x) {as.numeric(x[2]) - as.numeric(x[1])}) %>%
                        ungroup() %>% 
                        select(-c("Time_point", "ID")) %>% 
                        as.data.frame() %>% 
                        column_to_rownames(var = "SeqID")
  
  otu_table(ps.inst) <- otu_table(otu.tab.delta, taxa_are_rows = FALSE)
  
  pss.cor.delta[[i.ps]] <- ps.inst %>% 
                     prune_samples(sample_data(ps.inst)$Time_point == "T1", .)

}

# Run correlations 
#-------------------------------------------------------------------------------
strata.vec <- c("phenotype_diet", "diet_name", "diet_optimal", "all")

corr.res.dd <- NULL

ord.delta.metabl <-  delta.metabol.ciw1[sample_names(pss.cor.delta[[1]]), ]
  
for (i.strat in strata.vec) {
  
  if(i.strat == "all") {i.strat <- NULL}
  
  corr.res.dd <- phy_corr_matrix(phyloseqs_named_list = pss.cor.delta, 
                                  second_df = ord.delta.metabl, 
                                  cut_offs_vector = c(0.5, 0.75), 
                                  by_group = i.strat, 
                                  corr_method = "spearman") %>% 
                    mutate(Strata2 = i.strat) %>% 
                    bind_rows(corr.res.dd, .)
  
}


write.csv(corr.res.dd, "out/tables/corr_res_delta_delta.csv")
```

## Correlation: Results: T1 taxa and delta parameters 

### q-value <= 0.1

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.tab <- corr.res.T1[corr.res.T1$qvalue <= 0.1, ] %>% 
              as.data.frame()

kable(res.tab)
```

### Top 50 taxa by estimate 

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.tab <- corr.res.T1[order(abs(corr.res.T1$estimate), decreasing = TRUE), ] %>% 
                        .[1:50, ] %>% 
                        as.data.frame(.)
kable(res.tab)
```

## Correlation: Results: delta taxa and delta parameters 

### q-value <= 0.1

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.tab <- corr.res.dd[corr.res.dd$qvalue <= 0.1, ] %>% 
              as.data.frame()

kable(res.tab)
```

### Top 50 taxa by estimate 

```{r , message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
res.tab <- corr.res.dd[order(abs(corr.res.dd$estimate), decreasing = TRUE), ] %>% 
                      .[1:50, ] %>% 
           as.data.frame() 

kable(res.tab)
```
